packages/updatePrescriptionStatus/tests/testUpdatePrescriptionStatus.test.ts

```diff
@@ -0,0 +1,121 @@
+    const responseEntries: Array<BundleEntry> = []
+
+    const result = getXRequestID(event as APIGatewayProxyEvent, responseEntries)
+
+    expect(result).toEqual(X_REQUEST_ID)
+    expect(responseEntries.length).toEqual(0)
+  })
+
+  it.each([
+    {
+      event: {headers: {"x-request-id": ""}} as unknown,
+      scenarioDescription: "when event has empty x-request-id, return undefined and a response entry"
+    },
+    {
+      event: {headers: {}} as unknown,
+      scenarioDescription: "when event has a missing x-request-id, return undefined and a response entry"
+    }
+  ])("$scenarioDescription", async ({event}) => {
+    const responseEntries: Array<BundleEntry> = []
+
+    const result = getXRequestID(event as APIGatewayProxyEvent, responseEntries)
+
+    expect(result).toEqual(undefined)
+    expect(responseEntries.length).toEqual(1)
+    expect(responseEntries[0]).toEqual(badRequest("Missing or empty x-request-id header."))
+  })
+})
+
+describe("Unit test castEventBody", () => {
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(DEFAULT_DATE)
+  })
+
+  it("when body doesn't have correct resourceType and type, return undefined and a response entry", async () => {
+    const body = {resourceType: "NotBundle", type: "not_transaction"}
+    const responseEntries: Array<BundleEntry> = []
+
+    const result = castEventBody(body, responseEntries)
+
+    expect(result).toEqual(undefined)
+    expect(responseEntries.length).toEqual(1)
+    expect(responseEntries[0]).toEqual(badRequest("Request body does not have resourceType of 'Bundle' and type of 'transaction'."))
+  })
+
+  it("when body has correct resourceType and type, return bundle and no response entries", async () => {
+    const body = {resourceType: "Bundle", type: "transaction"}
+    const responseEntries: Array<BundleEntry> = []
+
+    const result = castEventBody(body, responseEntries)
+
+    expect(result).toBeDefined()
+    expect(responseEntries.length).toEqual(0)
+  })
+})
+
+describe("Unit test validateEntries", () => {
+  it("when a single entry is valid, returns true with a response in the response bundle", async () => {
+    mockValidateEntry.mockReturnValue({valid: true, issues: undefined})
+
+    const requestEntries = [{resource: {}, fullUrl: "valid"}] as Array<BundleEntry>
+    const responseEntries: Array<BundleEntry> = []
+
+    const result = validateEntries(requestEntries, responseEntries)
+
+    expect(result).toEqual(true)
+    expect(responseEntries.length).toEqual(1)
+
+    const validResponseEntry = responseEntries[0]
+    expect(validResponseEntry.fullUrl).toEqual("valid")
+    expect(validResponseEntry.response?.status).toEqual("200 Accepted")
+  })
+
+  it("when one of two entries is invalid, returns false with two responses in the response bundle", async () => {
+    mockValidateEntry.mockReturnValueOnce({valid: true, issues: undefined})
+    mockValidateEntry.mockReturnValueOnce({valid: false, issues: "issues"})
```

Comment:
The fact that these get queued up is a bit unintuitive. Could we make the mock check the input instead to make it more explicit?

Reply:
Sorted.